lib/core/src/main/java/io/atlasmap/core/DefaultAtlasContext.java

```diff
@@ -167,7 +191,18 @@ protected synchronized void init() throws AtlasException {
                     , this.atlasMappingUri);
             return;
         }
-
+        AtlasMapping atlasMapping = this.admHandler.getMappingDefinition();
+        String version = factory.getProperties().get(AtlasContextFactory.PROPERTY_ATLASMAP_CORE_VERSION);
+        String mappingVersion = atlasMapping.getVersion();
+        if (!validateVersion(version, mappingVersion)) {
+            throw new AtlasException(
```

Comment:
Can we only do `LOG.error(msg, e)` but not throwing it? maybe with saying like `The version "n.n.n" of Mapping Definition was detected. It might not work as expected with the runtime version "m.m.m"`

Reply:
And can we add some test cases, would be in DefaultAtlasContextTest